notebooks/search-analysis/search_analysis.ipynb

```diff
@@ -0,0 +1,399 @@
+    "query_sample['top_response']=query_sample['search_term'].map(lambda x:query_search_api(x,5))\n",
+    "\n",
+    "# Futz around with json fields\n",
+    "query_sample = query_sample.explode('top_response')\n",
+    "query_sample['description'] = sample['top_response'].map(lambda x:x.get('description',np.nan) if not isinstance(\n",
+    "                x, float) else [])\n",
+    "query_sample['id'] = sample['top_response'].map(lambda x:x.get('_id',np.nan) if not isinstance(\n",
+    "                x, float) else [])\n",
+    "query_sample['rank'] = sample['top_response'].map(lambda x:x.get('original_rank',np.nan) if not isinstance(\n",
+    "                x, float) else [])\n",
+    "query_sample= query_sample.reset_index(drop=True)\n",
+    "\n",
+    "\n",
+    "from ast import literal_eval\n",
+    "\n",
+    "relevancy_judgments = annotate(query_sample.index,\n",
+    "                               options=['very relevant','somewhat relevant','not relevant','no good answer'],\n",
+    "                               display_fn=lambda idx : display(sample.loc[idx,['search_term','description','id']]))\n",
+    "\n",
+    "query_sample['relevancy']= relevancy_judgments\n",
+    "\n",
+    "query_sample['relevancy'] =  query_sample['relevancy'].map(lambda x: literal_eval(x) if not isinstance(x, float) else np.nan)\n",
+    "score = {'not relevant':0,'somewhat relevant':1,'very relevant':2,'no good answer':np.nan}\n",
+    "query_sample['score'] = query_sample['relevancy'].map(lambda x: score[x[1]])\n",
+    "\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "relevancy_judgements"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#precision at k (k=5)\n",
+    "no_bestbets = relevancy_judgements[relevancy_judgements['best_bet']==False]\n",
+    "precision = no_bestbets[['query_length','session_id','search_term','score']][no_bestbets['score']>1].groupby(['search_term','session_id','query_length']).count()\n",
+    "precision['score'] = precision['score'] / 5\n",
+    "precision = precision.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Avg Precision at 5 \n",
+    "\n",
+    "from scipy.stats import sem, t\n",
+    "from scipy import mean\n",
+    "avg_score = precision[['query_length','score']].groupby('query_length').agg(['mean','size','sem'])\n",
+    "avg_score = avg_score.droplevel(0,1)\n",
+    "avg_score['h'] = avg_score['sem'] * t.ppf((1 + 0.95) / 2, avg_score['size'] - 1)\n",
+    "avg_score['lower'] = avg_score['mean'] - avg_score['h']\n",
+    "avg_score['upper'] = avg_score['mean'] + avg_score['h']\n",
+    "avg_score = avg_score.reset_index()\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
```

Comment:
Already called the import of the `matplotlib` package.
```suggestion

```

Reply:
gone!